library/core/src/macros/mod.rs

```diff
@@ -1273,6 +1273,19 @@ pub(crate) mod builtin {
     /// first macro invocation leading up to the invocation of the `file!`
     /// macro.
     ///
+    /// The file name is derived from the root module's source path passed to the Rust compiler
```

Comment:
@epage ...So, looking on a new day with clearer eyes, I think 80% of my confusion was just the usage of "root module" instead of
```suggestion
    /// The file name is derived from the crate's source path passed to the Rust compiler
```

as I believe "crate" is indeed the technically correct thing to say, here? But I am not inclined to argue much here if you think "root module" is preferable, because of the common confusion with "package" that we both know well.

Reply:
how about this?
```suggestion
    /// The file name is derived from the crate root's source path passed to the Rust compiler
```

I think "crate root" is the least ambiguous and most clear description